src/handlers/psp.ts

```diff
@@ -166,17 +161,36 @@ const initializeState = (config: Config): State => {
 
 const executeApiCalls = async (state: State): Promise<State> => {
   const { config, baseLogOptions, groupedSubscriptions } = state;
+  const airnodeHDNode = ethers.utils.HDNode.fromMnemonic(config.nodeSettings.airnodeWalletMnemonic);
+  const airnodeAddress = (
+    config.airnodeXpub
+      ? ethers.utils.HDNode.fromExtendedKey(config.airnodeXpub).derivePath('0/0')
+      : airnodeHDNode.derivePath(ethers.utils.defaultPath)
+  ).address;
+  if (config.airnodeAddress && config.airnodeAddress !== airnodeAddress) {
+    throw new Error(`xpub does not belong to Airnode: ${airnodeAddress}`);
+  }
```

Comment:
> One last thing to notice is that node.callApi() only uses airnodeAddress for regular requests so it might even be unnecessary to even set this value.

I'll look into this as it may need some changes on the airnode side :thinking: 

> Another thing to consider is that subscription objects also have airnodeAddress as prop.
> Then we can just do const airnodeAddress = state.airnodeWallet.address

Oherwise I can do either of these